packages/core/nuxt-theme-module/theme/components/AppHeader.vue

```diff
@@ -65,21 +65,32 @@ export default {
 
     const accountIcon = computed(() => isAuthenticated.value ? 'profile_fill' : 'profile');
 
-    const onAccountClicked = () => {
-      isAuthenticated && isAuthenticated.value ? root.$router.push('/my-account') : toggleLoginModal();
+    const handleSidebarOpen = async () => {
+      await loadCart();
+      toggleCartSidebar();
+    };
+
+    const handleAccountClick = async () => {
+      await loadUser();
```

Comment:
same here, you can't assume that its the only place account is entered from, this should work on every occasion when we navigate to user profile. 

Both auth check and loading user should be handled in my account page